From 96a3df4abd0282fa9e9f489a3b5c1d3f409b961d Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Fri, 27 Nov 2020 10:20:43 +0800
Subject: [PATCH] PCI: rockchip: dw: Move rk_pcie_establish_link() into
 rk_pcie_add_ep()

rk_pcie_establish_link() will be called by dwc core driver for RC,
so we only need it for ep function.

Change-Id: I278f3749bc0571037192ff0c4d03ba306c453316
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/pci/controller/dwc/pcie-dw-rockchip.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/pci/controller/dwc/pcie-dw-rockchip.c b/drivers/pci/controller/dwc/pcie-dw-rockchip.c
index 82ce9cabbdb8..6831897eaeaa 100644
--- a/drivers/pci/controller/dwc/pcie-dw-rockchip.c
+++ b/drivers/pci/controller/dwc/pcie-dw-rockchip.c
@@ -682,6 +682,12 @@ static int rk_pcie_add_ep(struct rk_pcie *rk_pcie)
 
 	rk_pcie_ep_setup(rk_pcie);
 
+	ret = rk_pcie_establish_link(rk_pcie->pci);
+	if (ret) {
+		dev_err(dev, "failed to establish pcie link\n");
+		return ret;
+	}
+
 	rk_pcie->dma_obj = rk_pcie_dma_obj_probe(dev);
 	if (IS_ERR(rk_pcie->dma_obj)) {
 		dev_err(dev, "failed to prepare dma object\n");
@@ -1165,12 +1171,6 @@ static int rk_pcie_probe(struct platform_device *pdev)
 	rk_pcie->link_gen = of_pci_get_max_link_speed(np);
 	rk_pcie_set_gens(rk_pcie);
 
-	ret = rk_pcie_establish_link(pci);
-	if (ret) {
-		dev_err(dev, "failed to establish pcie link\n");
-		goto deinit_clk;
-	}
-
 	switch (rk_pcie->mode) {
 	case RK_PCIE_RC_TYPE:
 		ret = rk_add_pcie_port(rk_pcie);
-- 
2.35.3

